src/time-picker/time-picker.jsx

```diff
@@ -108,6 +108,17 @@ const TimePicker = React.createClass({
       muiTheme: this.context.muiTheme || getMuiTheme(),
     };
   },
+  
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newState = this.state;
+    if (nextContext.muiTheme) {
+      newState.muiTheme = nextContext.muiTheme; 
+    }
+    if (!Date.isEqualTime(this.state.time, nextProps.defaultTime)) {
+      newState.time = nextProps.defaultTime;
```

Comment:
of course, that would be better, the prop `value` is going to have precedence over `defaultTime` at first rendering, right?